src/resource/solidDataset.ts

```diff
@@ -827,6 +827,17 @@ export async function deleteContainer(
   }
 }
 
+function isChildResource(a: string, b: string): boolean {
+  const parent = new URL(b);
+  const child = new URL(a);
+  const isAncestor = child.href.startsWith(parent.href);
+  const urlDiff = child.href.substring(parent.href.length, child.href.length);
+  // The child path component that isn't present in the parent should only
+  // potentially include slashes at the end (if it is a container).
+  const isDirectChild = urlDiff.includes(urlDiff.replace("/", " ").trim());
+  return isAncestor && isDirectChild;
```

Comment:
The following returns `true` even though I wouldn't expect it to `isChildResource('http://a/b//c/d/e', 'http://a/b/')`.

To fix this I would do something equivalent to
```suggestion
  const stripped = urlDiff.replace(/(^\/)|(\/$)/g, "");
  return isAncestor && stripped.length >= 1 && !stripped.includes("/");
```

Reply:
I'd also be inclined to do this all based off `.path` rather than `.href` and check that all the other `URL` attributes are the same in order to prevent things like `http://example.org/a?q1=a/a` being considered to be contained in `http://example.org/a?q1=a/`

Reply:
b55a497ff4775001106d5eceebb1a4b3a770f833 addresses your first point (I didn't realize `.replace` only replaced the first occurrence for non-regexes), and 15dc7f566ece765ff36e87f40ab2bef9762142ee your second point, both very good observations.